homeassistant/components/linky/config_flow.py

```diff
@@ -0,0 +1,110 @@
+"""Config flow to configure the Linky integration."""
+import logging
+
+import voluptuous as vol
+from pylinky.client import LinkyClient, PyLinkyError
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_TIMEOUT, CONF_USERNAME
+
+from .const import DEFAULT_TIMEOUT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+_LOGGER.error("LINKY_CONFIG_FLOW")
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class LinkyFlowHandler(config_entries.ConfigFlow):
+    """Handle a Linky config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize Linky config flow."""
+        self._username = None
+        self.__password = None
+        self._timeout = None
+
+    def _configuration_exists(self, username: str) -> bool:
+        """Return True if username exists in configuration."""
+        for entry in self.hass.config_entries.async_entries(DOMAIN):
+            if entry.title == username:
+                return True
+        return False
+
+    async def _show_setup_form(self, user_input=None, errors=None):
+        """Show the setup form to the user."""
+
+        if user_input is None:
+            user_input = {}
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_USERNAME, default=user_input.get(CONF_USERNAME, "")
+                    ): str,
+                    vol.Required(
+                        CONF_PASSWORD, default=user_input.get(CONF_PASSWORD, "")
+                    ): str,
+                    vol.Required(
+                        CONF_TIMEOUT,
+                        default=user_input.get(CONF_TIMEOUT, DEFAULT_TIMEOUT),
+                    ): int,
+                }
+            ),
+            errors=errors or {},
+        )
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+
+        if user_input is None:
+            return await self._show_setup_form(user_input, None)
+
+        self._username = user_input[CONF_USERNAME]
+        self.__password = user_input[CONF_PASSWORD]
+        self._timeout = user_input[CONF_TIMEOUT]
+
+        if self._configuration_exists(self._username):
+            errors = {}
+            errors[CONF_USERNAME] = "username_exists"
+            return await self._show_setup_form(user_input, errors)
+
+        client = LinkyClient(self._username, self.__password, None, self._timeout)
+        try:
+            client.login()
+            client.fetch_data()
+        except PyLinkyError as exp:
+            _LOGGER.error(exp)
+            errors = {}
+            errors["PyLinkyError"] = exp
```

Comment:
This isn't how the errors dict interface works. Please see other config flows for examples and read the dev docs.

https://developers.home-assistant.io/docs/en/data_entry_flow_index.html#show-form

Reply:
Like I did [in ligne 73](https://github.com/home-assistant/home-assistant/pull/26076/files#diff-c8baca8e2c0d6726713dca9f016d3d76R73) ?

Or it's kinda the 3rd question I was asking ?

Reply:
Yes. Each error needs to be present in strings.json. The messages in errors dict are user facing, so the messages shouldn't contain debug information. Just enough information so the user can take appropriate action.

Reply:
The `exp` var is actually a string returning "Login error: Please check your username/password." in case of the network and Enedis service are working with fake credentials.

So it was to display this error, which is human readable, to the user. The problem is that Enedis and so Linky are French entities so the exception couldn't be translated that way.

Reply:
For now I will just parse the exception to display a translated error, then I will contribute to `pylinky`to proposed specific exceptions for each case so we can use this to build our feedbacks.

Reply:
PyLinky PR created : https://github.com/Pirionfr/pyLinky/pull/26